opensuse/leap/15.0/Dockerfile

```diff
@@ -1,6 +1,15 @@
 FROM opensuse/leap:15.0
 MAINTAINER Alexander V. Tikhonov <avtikhon@gmail.com>
 
+# Add the same repositories based on Yandex paths as default
+RUN for f in /etc/zypp/repos.d/* ; do \
+        r=$(basename $f | sed 's#.repo$##g') ; \
+        nf=`dirname $f`/${r}_yandex.repo ; \
+        cp $f $nf ; \
+        sed "s#$r#${r}_yandex#g" -i $nf ; \
+        sed 's#http://download.opensuse.org#https://mirror.yandex.ru/opensuse#g' -i $nf ; \
+    done
```

Comment:
This way we break uniqueness of repository name, which looks as requirement of the yum repositories format:

> **Example:** [repositoryid]
> name=Some name for this repository
> baseurl=url://path/to/repository/
>
> **repositoryid** Must be a unique name for each repository, one word. 

https://linux.die.net/man/5/yum.conf

The behaviour *may be* one that you want to achieve, but this setup is not guaranteed to be workable at all.

OTOH, there is a 'legal' way to achieve the same:

> **baseurl** Must be a URL to the directory where the yum repository's 'repodata' directory lives. Can be an http://, ftp:// > or file:// URL. You can specify multiple URLs in one baseurl statement. The best way to do this is like this:
> [repositoryid]
> name=Some name for this repository
> baseurl=url://server1/path/to/repository/
> url://server2/path/to/repository/
> url://server3/path/to/repository/

Don't miss the following note:

> If you list more than one baseurl= statement in a repository you will find yum will ignore the earlier ones and probably act bizarrely. Don't do this, you've been warned.

Reply:
Ouch, I see, you replace the `repositoryid` (it corresponds to the file name in fact, however it must not). So, we end up with two different repositories: one that is available and one that is not. Successful updating of one repository should not make a package manager (zypper) more tolerate to fail of updating another repository. So I anyway think that we should list several URLs in the `baseurl` field in order to overcome downtime of one of mirrors.

Reply:
Right, initial solution failed on unreachable original paths. Implemented your variant, all checks passed.

Reply:
Nice, thanks!